homeassistant/components/update/__init__.py

```diff
@@ -0,0 +1,297 @@
+"""Support for Update."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Awaitable, Callable, Coroutine
+import dataclasses
+import logging
+from types import ModuleType
+from typing import Any, Protocol
+
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import integration_platform, storage
+from homeassistant.helpers.typing import ConfigType
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "update"
+
+INFO_CALLBACK_TIMEOUT = 5
+STORAGE_VERSION = 1
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Update integration."""
+    hass.data[DOMAIN] = manager = UpdateManager(hass=hass)
+    await manager.load()
+
+    websocket_api.async_register_command(hass, handle_info)
+    websocket_api.async_register_command(hass, perform_update)
+    websocket_api.async_register_command(hass, handle_skip)
+
+    await integration_platform.async_process_integration_platforms(
+        hass, DOMAIN, _register_update_platform
+    )
+
+    return True
+
+
+async def _register_update_platform(
+    hass: HomeAssistant,
+    integration_domain: str,
+    platform: ModuleType,
+) -> None:
+    """Register an update platform."""
+    if hasattr(platform, "async_register"):
+        platform.async_register(UpdateRegistration(hass, integration_domain))
```

Comment:
Why not store `platform` ?

Reply:
Also please define a `Protocol` to type the platform. Example: https://github.com/home-assistant/core/blob/dev/homeassistant/components/diagnostics/__init__.py#L49-L60